File: app/src/main/java/com/cliff/conch/box/scene/ActivityThread.java
package com.cliff.conch.box.scene;

import com.cliff.conch.box.reflect.RefClass;
import com.cliff.conch.box.reflect.RefMethod;
import com.cliff.conch.box.reflect.RefStaticMethod;

public class ActivityThread {

    // 关键就在于Type的赋值，RefClass.load函数中遍历了当前ActivityThread的所有static值，并对其进行赋值，通过反射封装调用
    @SuppressWarnings("unused")
    public static Class<?> TYPE = RefClass.load(ActivityThread.class, "android.app.ActivityThread");
    public static RefStaticMethod<?> currentActivityThread;
    public static RefMethod<String> getProcessName;
}

File: reflect-android/src/main/java/reflect/android/app/ActivityThread.kt
package reflect.android.app

import android.content.pm.ApplicationInfo
import android.util.ArrayMap
import com.cliff.reflection.common.IReflect
import com.cliff.reflection.common.annotation.PField
import com.cliff.reflection.common.annotation.PMethod
import com.cliff.reflection.common.annotation.PMethodParameter
import com.cliff.reflection.common.annotation.PStaticField
import com.cliff.reflection.common.annotation.PStaticMethod
import com.cliff.reflection.common.annotation.ProxyClass
import reflect.android.content.res.CompatibilityInfo
import java.lang.ref.WeakReference

@ProxyClass("android.app.ActivityThread")
interface ActivityThread : IReflect {
    companion object


    @PStaticField("android.content.pm.IPackageManager")
    val sPackageManager: Any?

    @PStaticMethod("android.app.ActivityThread")
    fun currentActivityThread(): ActivityThread?


    @PField("android.app.ActivityThread\$ApplicationThread")
    val mAppThread: Any?

    @PField
    val mPackages: ArrayMap<String, WeakReference<Any>>

    @PField
    val mResourcePackages: ArrayMap<String, WeakReference<Any>>

    @PMethod
    fun getProcessName(): String?

    @PMethod("android.app.ActivityThread\$ApplicationThread")
    fun getApplicationThread(): Any

    @PMethod("android.app.LoadedApk")
    fun getPackageInfo(
        aInfo: ApplicationInfo?,
        @PMethodParameter("android.content.res.CompatibilityInfo") compatInfo: CompatibilityInfo?,
        baseLoader: ClassLoader? = null,
        securityViolation: Boolean = false,
        includeCode: Boolean = true,
        registerPackage: Boolean = false,
        isSdkSandbox: Boolean = false
    ): Any

}